src/components/Composer/Comment/New.tsx

```diff
@@ -200,29 +207,62 @@ const NewComment: FC<Props> = ({ publication }) => {
     if (!currentProfile) {
       return toast.error(SIGN_WALLET);
     }
+
+    if (isAudioComment) {
+      setCommentContentError('');
+      const AudioPublicationSchema = z.object({
+        title: z.string().trim().min(1, { message: 'Invalid audio title' }),
+        author: z.string().trim().min(1, { message: 'Invalid author name' }),
+        cover: z.string().trim().min(1, { message: 'Invalid cover image' })
+      });
+      const parsedData = AudioPublicationSchema.safeParse(audioPublication);
+      if (!parsedData.success) {
+        const issue = parsedData.error.issues[0];
+        return setCommentContentError(issue.message);
+      }
+    }
+
     if (publicationContent.length === 0 && attachments.length === 0) {
       return setCommentContentError('Comment should not be empty!');
     }
 
     setCommentContentError('');
     setIsUploading(true);
+
+    const attributes = [];
+    if (isAudioComment) {
+      attributes.push({
+        traitType: 'author',
+        displayType: 'string',
+        value: audioPublication.author
+      });
+    }
+
     const id = await uploadToArweave({
       version: '2.0.0',
       metadata_id: uuid(),
       description: trimify(publicationContent),
       content: trimify(publicationContent),
       external_url: `https://lenster.xyz/u/${currentProfile?.handle}`,
-      image: attachments.length > 0 ? attachments[0]?.item : null,
-      imageMimeType: attachments.length > 0 ? attachments[0]?.type : null,
-      name: `Comment by @${currentProfile?.handle}`,
+      image: attachments.length > 0 ? (isAudioComment ? audioPublication.cover : attachments[0]?.item) : null,
+      imageMimeType:
+        attachments.length > 0
+          ? isAudioComment
+            ? audioPublication.coverMimeType
+            : attachments[0]?.type
+          : null,
```

Comment:
**suggestion:** better move to switch case if the conditions are growing, we need to support video as well in future 🙇🏼